src/validation.cpp

```diff
@@ -361,10 +361,10 @@ bool TestLockPointValidity(const LockPoints* lp)
     return true;
 }
 
-bool CheckSequenceLocks(const CTransaction &tx, int flags, LockPoints* lp, bool useExistingLockPoints)
+bool CheckSequenceLocks(const CTxMemPool& pool, const CTransaction& tx, int flags, LockPoints* lp, bool useExistingLockPoints)
 {
     AssertLockHeld(cs_main);
```

Comment:
Unrelated to this PR, but couldn't this `AssertLockHeld` be removed?

Reply:
Good point, but I think we want to keep these for now, since not all compilers support these compile-time annotations and sometimes they have to be disabled due to shortcomings.

Reply:
On the other hand, the annotation is incomplete, missing pool.cs?

Reply:
True, but should be done in a separate pull, since the changes required are non-trivial (more than one line)

Reply:
In order to add that annotation `txmempool.h` must be included in `validation.h` and I'm not sure if that is correct because of the circular dependency `"txmempool -> validation -> txmempool"`. I don't know what is the plan here but I think mempool should not depend on validation or am I wrong?